types/encoding.go

```diff
@@ -103,6 +119,31 @@ func (c Currency) MarshalSia(w io.Writer) error {
 	return encoding.WritePrefix(w, c.i.Bytes())
 }
 
+// MarshalSiaSize returns the encoded size of c.
+func (c Currency) MarshalSiaSize() int {
+	// from math/big/arith.go
+	const (
+		_m    = ^big.Word(0)
+		_logS = _m>>8&1 + _m>>16&1 + _m>>32&1
+		_S    = 1 << _logS // number of bytes per big.Word
+	)
+
+	// start with the number of Words * number of bytes per Word, then
+	// subtract trailing bytes that are 0
+	bits := c.i.Bits()
+	size := len(bits) * _S
```

Comment:
I also tried calculating the size like so:
```go
bitlen := c.i.BitLen()
size := (bitlen / _S)
if bitlen%_S != 0 {
	size++
}
return 8 + size
```
but surprisingly, this is slower (10ns vs 6ns), and the slowness isn't caused by the modulo. I guess `BitLen` does needless work calculating the exact number of bits, when really we just care about the number of bytes.